Add builder method with_reliable_init()

vcrn opened the pull request:
This PR should somewhat fix issue #4, at least make a workaround available.

This PR adds a new builder method, `with_reliable_init()`, which toggles the screen off and on 3 times, with a user-set delay in between each toggle. I tried creating a workaround that only used a pause, without toggling the screen, but was unable to improve the system's behavior.

I see this as more of a workaround than as a solution, but a workaround that has been very useful for me. Because of that, I suggest we leave issue #4 open.

mjhouse commented on src/display.rs:
```diff
@@ -405,6 +405,31 @@ where
+    /// Note that this method should be run as close as possible to
+    /// `.build()`, and that it leaves LCD with Display::On.
+    ///
+    /// let mut lcd: LcdDisplay<_,_> = LcdDisplay::new(rs, en, delay)
```
I think you need three backticks before the example as well as after, and make sure it builds and runs. Refer to [the rust lang docs](https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html) for more info on doc tests.

mjhouse replied:
Actually, I don't think any of the other examples run, so don't worry about that- we'll handle it in a different issue. I think you are missing three backticks though.

vcrn replied:
Ah, yes, forgot about the backticks. I'm quite sure this one is able to run, but I'll make sure to double check

mjhouse commented:
Also, how does this interact with the `with_display(mut self, value: Display)`? If you call this before you call with_display, that seems ok (this function will turn the screen on and then with_display will turn it back off if they pass Display::Off), but if you call it *after* with_display, this function will turn the display *on*, regardless of whether they passed Display::Off or not.

vcrn commented:
Thank you for the feedback! I've fixed both of your points. Regarding whether the example runs: with the correct set up (setting the pins, etc, like the examples in `examples/`) it does run.

Regarding the method always leaving the LCD::On, you are correct. I tried to be clear about it in the doc, but was unhappy with this behavior. I've added an if-state to check whether it's On or Off, and then toggle it accordingly to leave it in the same state. Do you think it's enough to check whether it's On and then do an `else` or should an `else if` be used to check whether it's off? It seems like to only two states possible, I just want to be sure that I'm not missing something.

mjhouse commented:
That should work. It's a little verbose, but I can't think of a way to simplify it off the top of my head. I'll merge it.

vcrn commented:
Yeah, it's quite unelegant. I'll probably open a PR for something nicer later on, unless the issue is fixed before then without using this builder method.

It would probably be nicer to save the state of the display to a variable, then run the loop that was present in my first commit (without taking into consideration the state of the display), and then set the display's state to that of the variable's, and then returning.
